src/main/java/school/hei/patrimoine/visualisation/swing/ihm/google/GoogleDocsSubmitScreen.java

```diff
@@ -156,57 +113,58 @@ private void loadDataInBackground() {
     loadingDialog.setSize(300, 100);
     loadingDialog.setLocationRelativeTo(inputFrame);
 
-    SwingWorker<List<Patrimoine>, Void> worker =
+    SwingWorker<Object, Void> worker =
         new SwingWorker<>() {
           @Override
-          protected List<Patrimoine> doInBackground() {
-            var ids = extractInputIds();
-            List<String> codePatrimoinesVisualisables = new ArrayList<>();
-            for (var id : ids) {
-              var code = googleApi.readDocsContent(authDetails, id);
-              codePatrimoinesVisualisables.add(code);
-            }
-            List<Patrimoine> patrimoinesVisualisables = new ArrayList<>();
-            PatrimoineCompiler patrimoineCompiler = new PatrimoineCompiler();
-            for (String codePatrimoine : codePatrimoinesVisualisables) {
-              String className = new ClassNameExtractor().apply(codePatrimoine);
-
-              Patrimoine patrimoineVisualisable =
-                  patrimoineCompiler.apply(className, codePatrimoine);
-
-              patrimoinesVisualisables.add(patrimoineVisualisable);
-            }
-            return patrimoinesVisualisables;
+          protected List<NamedLink> doInBackground() {
+            return extractInputData();
           }
 
           @Override
           protected void done() {
             loadingDialog.dispose();
             try {
-              final List<Patrimoine> patrimoinesVisualisables = get();
-              openResultFrame(patrimoinesVisualisables);
+              final List<NamedLink> docsLink = (List<NamedLink>) get();
+              openResultFrame(docsLink);
             } catch (InterruptedException | ExecutionException e) {
               throw new RuntimeException(e);
             }
           }
         };
 
-    // Show the loading dialog and start the worker
     worker.execute();
-    loadingDialog.setVisible(true); // This will block until loadingDialog is disposed
+    loadingDialog.setVisible(true);
   }
 
-  private List<String> extractInputIds() {
-    List<String> ids = new ArrayList<>();
-    for (JTextField field : inputFields) {
-      String rawLink = field.getText();
-      var parsedId = linkIdParser.apply(rawLink);
-      ids.add(parsedId);
+  private List<NamedLink> extractInputData() {
+    List<NamedLink> linkDataList = new ArrayList<>();
+
+    String rawText = inputFields.getFirst().getText();
```

Comment:
this code will be corrected as `rawTest = inputField.getText()`

Reply:
done